src/download-database.ts

```diff
@@ -0,0 +1,187 @@
+// This file borrows heavily from the actions "downloadTool" function:
+// https://github.com/actions/toolkit/blob/main/packages/tool-cache/src/tool-cache.ts#L38
+
+import * as fs from "fs";
+import * as path from "path";
+import * as stream from "stream";
+import * as util from "util";
+
+import * as core from "@actions/core";
+import * as httpm from "@actions/http-client";
+import { IHeaders } from "@actions/http-client/interfaces";
+import * as io from "@actions/io";
+
+export class HTTPError extends Error {
+  httpStatusCode: number | undefined;
+  httpMessage: string;
+  constructor(httpStatusCode: number | undefined, httpMessage: string) {
+    super(`Unexpected HTTP response: ${httpStatusCode}. ${httpMessage}`);
+    // Set status code and error message to `this`
+    this.httpStatusCode = httpStatusCode;
+    this.httpMessage = httpMessage;
+  }
+}
+
+const userAgent = "actions/tool-cache";
+
+/**
+ * Download a database from an url and stream it into a file
+ *
+ * @param url       url of database to download
+ * @param dest      path to download database
+ * @param auth      authorization header
+ * @param headers   other headers
+ * @returns         path to downloaded database
+ */
+export async function downloadDatabaseFile(
+  url: string,
+  dest: string,
+  auth?: string
+): Promise<string> {
+  await io.mkdirP(path.dirname(dest));
+  core.debug(`Downloading ${url}`);
+  core.debug(`Destination ${dest}`);
+
+  const maxAttempts = 3;
+  const minSeconds = 10;
+  const maxSeconds = 20;
+  const retryHelper = new RetryHelper(maxAttempts, minSeconds, maxSeconds);
+  return await retryHelper.execute(
+    async () => {
+      return await downloadDatabaseFileAttempt(url, dest || "", auth);
```

Comment:
`dest` is always defined, so `dest || ""` is unnecessary. 
```suggestion
      return await downloadDatabaseFileAttempt(url, dest, auth);
```